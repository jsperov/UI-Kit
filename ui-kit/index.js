import React from 'react';
import ReactDOM from 'react-dom';

import { StarRating } from './star/index.jsx'

ReactDOM.render(
  <StarRating />,
  document.getElementById('react-ui-kit'),
);

