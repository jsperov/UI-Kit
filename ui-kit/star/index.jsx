import React from 'react';

import { Star } from './star';
import { StarBox } from './star.style';

class StarRating extends React.Component {
  static defaultProps = {
    repeat: 5,
    color: '#f9f',
    activeColor: '#e7eb07',
    activeStar: 0
  }

  state = {
    rating: 0
  };

  onClick = rating => this.setState({ rating })

  renderStar = (repeat) => [...Array(repeat)].map((item, rate) => 
    <Star
      rate={rate}
      selected={this.state.rating > rate}
      onClick={() => this.onClick(rate + 1)}
      color={this.props.color}
      activeColor={this.props.activeColor}
    />
  );

  render() {
    const { repeat, color, activeColor } = this.props;
    const { rating } = this.state;

    return (
      <StarBox
        color={color}
        activeColor={activeColor}
      >
        Звезды {`${rating} из  ${repeat}`}
        {this.renderStar(repeat)}
      </StarBox>
    )
  }
}

export { StarRating }
