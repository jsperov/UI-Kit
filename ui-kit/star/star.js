import React from 'react';

const Star = ({
  rate,
  selected,
  color,
  activeColor,
  ...rest
}) => 
  <div
    className={`star ${selected ? 'active' : ''}`}
    style={
      {
        background: selected ? activeColor : color,
      }
    }
    {...rest}
  />

export { Star }