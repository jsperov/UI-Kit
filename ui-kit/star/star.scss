.star {
  display: inline-block;
  margin: 0 7px;
  width: 25px;
  min-height: 25px;
  background: #e7eb078a;
  -webkit-clip-path: polygon(
    0% 50%,
    35% 35%,
    50% 0%,
    65% 35%,
    100% 50%,
    65% 65%,
    50% 100%,
    35% 65%
  );
  cursor: pointer;
}

.star.active {
  background: #e7eb07;
}