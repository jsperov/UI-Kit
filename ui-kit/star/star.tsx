import * as React from 'react';

type Props = {
  selected: boolean,
  onClick?: () => void,
}

const Star: React.SFC<Props> = ({
  selected=0,
  ...rest
}) => 
  <React.Fragment>
    <div
      className={`star ${selected ? 'active' : ''}`}
      {...rest}
    />
  </React.Fragment>

export { Star }