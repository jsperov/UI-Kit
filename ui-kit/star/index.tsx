import * as React from 'react';


import { Star } from './star';
import { StarBox } from './star.style';

type IProps = {
  repeat: number,
  color: string,
  activeColor: string,
  activeStar: string
}

type State = {
  rating: number
};

class StarRating extends React.Component<IProps, State> {
  static defaultProps = {
    repeat: 5,
    color: '#f9f',
    activeColor: '#e7eb07',
    activeStar: 0
  }

  state = {
    rating: 0
  };

  componentDidMount() {
    //TODO: load rating service
  }

  onClick = (rating: number) : void => this.setState({ rating })

  renderStar = (repeat: number) : JSX.Element[] => [...new Array(repeat).fill(null)].map((item, rate) => 
    <Star
      selected={this.state.rating > rate}
      onClick={() => this.onClick(rate + 1)}
      color={this.props.color}
      activeColor={this.props.activeColor}
    />
  );

  render() {
    const { repeat, color, activeColor } = this.props;
    const { rating } = this.state;

    return (
      <StarBox
        color={color}
        activeColor={activeColor}
      >
        Звезды {`${rating} из  ${repeat}`}
        {this.renderStar(repeat)}
      </StarBox>
    )
  }
}

export { StarRating }
