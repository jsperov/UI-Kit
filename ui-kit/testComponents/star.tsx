import * as React from 'react';

import { StarRating } from '../star';

import * as imageStar from './star.png';

class App extends React.Component {
  state = {
    rating: 0
  }

  onClick = (rating: number) : void => this.setState({ rating })

  render() {
    const { rating } = this.state

    return(
      <StarRating
        repeat={5}
        rating={rating}
        onClick={this.onClick}
        image={imageStar}
      />
    ) 
  }
}

export { App }
