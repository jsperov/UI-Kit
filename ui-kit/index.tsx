import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { App } from './testComponents/star'

import { fn } from './example'

console.log('fn - ', fn('dsads?a=3&b=3', ['b']))

ReactDOM.render(
  <App />,
  document.getElementById('react-ui-kit'),
);

