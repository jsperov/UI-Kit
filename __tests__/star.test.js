import {shallow} from 'enzyme';
import {StarRating} from '../ui-kit/star';
import toJson from 'enzyme-to-json';

describe('StarBox', () => {
  it('should render correctly', () => {
    const output = shallow(
      <StarRating
        repeat={5}
        rating={1}
        onClick={() => this.setState({ rating: 5})}
      />
    )
    
    expect(toJson(output)).toMatchSnapshot();
  });
});