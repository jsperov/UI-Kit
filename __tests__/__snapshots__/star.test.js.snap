// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarBox should render correctly 1`] = `
<styled.div
  activeColor="#e7ab07"
  color="#e7eb07"
  image={null}
>
  Звезды 1 из 5
  <Star
    image={null}
    key="key-0"
    onClick={[Function]}
    selected={true}
  />
  <Star
    image={null}
    key="key-1"
    onClick={[Function]}
    selected={false}
  />
  <Star
    image={null}
    key="key-2"
    onClick={[Function]}
    selected={false}
  />
  <Star
    image={null}
    key="key-3"
    onClick={[Function]}
    selected={false}
  />
  <Star
    image={null}
    key="key-4"
    onClick={[Function]}
    selected={false}
  />
</styled.div>
`;
